### PR TITLE
syncer:  add timeout mechanism for ddl/dml execution

### PR DESCRIPTION
**Pull request by GregoryIan**
* make dml and ddl use different connections

### CODE REVIEW

File: syncer/syncer.go [hunk 1]
```diff
@@ -825,6 +826,36 @@ func (s *Syncer) getBinlogStreamer() (*replication.BinlogStreamer, bool, error)
 	return s.startSyncByPosition()
 }
 
+// toDBs[workerCountunt] for quering db table structure
+// toDBs[workerCount+1] for executing ddl
```
Comment (holys) [inline syncer/syncer.go]: Comment seems incorrect.

File: syncer/syncer.go [hunk 2]
```diff
@@ -825,6 +826,36 @@ func (s *Syncer) getBinlogStreamer() (*replication.BinlogStreamer, bool, error)
...
+		return errors.Trace(err)
+	}
+
+	s.toDBs = make([]*sql.DB, 0, s.cfg.WorkerCount+2)
+	s.toDBs, err = createDBs(s.cfg.To, s.cfg.WorkerCount, maxDMLConnectionTimeout)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// db for ddl
+	ddlDB, err := createDB(s.cfg.To, maxDDLConnectionTimeout)
```
Comment (holys) [inline syncer/syncer.go]: I prefer a separated ddlDB, which seems clearer. So as the queryDB.

**Comment by holys** [comment]
LGTM

**Comment by nolouch** [comment]
LGTM

**Pull request merged**